### PR TITLE
promote baselinerules submodule

### PR DESCRIPTION
**Pull request by shireenf-ibm**

### CODE REVIEW

**Comment by shireenf-ibm** [comment]
Hi @adisos,
I found out that the "debug/ print" problem occurs in [src/baseline_verify.py](https://github.com/np-guard/baseline-rules-verifier/blob/master/src/baseline_verify.py) line 98
`elif nca_run.returncode != 0:
                details = '\n\n'.join(str(nca_run.stdout).split('\n')[2:5])`

this ignores the first two lines in the nca's stdout (`[2:5]`) which should be both similar to the form of : `global: cluster has 12 unique endpoints, 1 namespaces`
The old version of nca , always printed this stats sentence twice (once for the "global" topology and the second for the "base" topology, since it was built twice even when only "global" params are provided (ns_list and pod_list))
In the new version of nca, if the base topology is not provided with "base_" params a copy of the global peer container is used and the "stats" messages is not printed again (that is why in this PR , the first line of the query output was actually the second of nca.stdout and then was not printed here).
In order to fix this problem , **I am openning a new PR in the nca repository , which will print the peerContainer stats msg twice again.**
after merging the nca's PR, this should run successfully

**Comment by shireenf-ibm** [comment]
> > 
> 
> Hi @shireenf-ibm , good catch. maybe we should not duplicate the this stats sentence in nca, and update the code in baseline verifier to ignore either line 1 or lines 1-2, depending on which is on the format of this sentence?

I thought of that  too but since I don't know if other repositories also use nca and expects the same output
I decided to make a change in the nca.
would like to know your opinion

**Comment by adisos** [comment]
> > > 
> > 
> > 
> > Hi @shireenf-ibm , good catch. maybe we should not duplicate the this stats sentence in nca, and update the code in baseline verifier to ignore either line 1 or lines 1-2, depending on which is on the format of this sentence?
> 
> I thought of that too but since I don't know if other repositories also use nca and expects the same output I decided to make a change in the nca. would like to know your opinion

We do not rely on this output anywhere else.. I think it might be better to avoid this duplication in the output if it is redundant.

File: src/baseline_verify.py [hunk 1]
```diff
@@ -95,7 +94,12 @@ def verify(self, args):
             if args.debug is not None:
                 details = nca_run.stdout + '\n' + nca_run.stderr
             elif nca_run.returncode != 0:
-                details = '\n\n'.join(str(nca_run.stdout).split('\n')[2:5])
+                topology_output_words = ['cluster has', 'unique endpoints', 'namespaces']
+                nca_stdout = str(nca_run.stdout).split('\n')
+                if all(word in nca_stdout[1] for word in topology_output_words):
```
Comment (adisos) [inline src/baseline_verify.py:99]:
```suggestion
                # Remove nca's output about cluster topology info to set details output
                if all(word in nca_stdout[1] for word in topology_output_words):
```

**Review by adisos — COMMENTED** [review]
Hi @shireenf-ibm , looks good, just added some comment for documentation.

**Review by adisos — APPROVED** [review]

**Pull request merged**